examples/shaders/compute-shader/src/lib.rs

```diff
@@ -1,17 +1,31 @@
-#![cfg_attr(
-    target_arch = "spirv",
-    no_std,
-    feature(register_attr),
-    register_attr(spirv)
-)]
+#![no_std]
+#![cfg_attr(target_arch = "spirv", feature(register_attr), register_attr(spirv))]
 // HACK(eddyb) can't easily see warnings otherwise from `spirv-builder` builds.
 #![deny(warnings)]
 
 extern crate spirv_std;
 
+use glam::UVec3;
 #[cfg(not(target_arch = "spirv"))]
 use spirv_std::macros::spirv;
 
-// LocalSize/numthreads of (x = 32, y = 1, z = 1)
-#[spirv(compute(threads(32)))]
-pub fn main_cs() {}
+// Adapted from the wgpu hello-compute example
+
+pub fn collatz(mut n: u32) -> u32 {
+    let mut i = 0;
+    while n != 1 {
+        n = if n % 2 == 0 { n / 2 } else { (3 * n) + 1 };
+        i += 1;
+    }
+    i
+}
+
+// LocalSize/numthreads of (x = 1, y = 1, z = 1)
```

Comment:
it would be great to increase this for perf. Probably something like (x = 16, y = 1, z = 1) (or maybe 32) as we'll eventually be ALU bound for high enough n.